lib/galaxy/model/__init__.py

```diff
@@ -6200,18 +6199,26 @@ def attribute_columns(column_collection, attributes, nesting_level=None):
             .add_columns(*attribute_columns(DatasetPermissions, dataset_permission_attributes))
         )
         for entity in return_entities:
-            q = q.add_entity(entity)
+            q = q.add_columns(entity)
             if entity == DatasetCollectionElement:
                 q = q.filter(entity.id == dce.c.id)
-        return q.distinct().order_by(*order_by_columns)
+
+        q = q.distinct().order_by(*order_by_columns)
+        # With DISTINCT, all columns that appear in the ORDER BY clause must appear in the SELECT clause.
+        q = q.add_columns(*order_by_columns)
```

Comment:
I worry about this, in terms of correctness but also ease of use. Not that `_get_nested_collection_attributes` was ever very clean, but at least it was relatively straightforward to use. Could you turn this into a subquery and then select only the columns we care about, excluding the ones we added for the sake of the distinct statement ?
Or alternatively only apply the distinct statement based on a flag that the caller sets ? I think we only need the distinct for e.g dataset_states_and_extensions_summary